cdap-common/src/test/java/co/cask/cdap/common/zookeeper/coordination/ResourceCoordinatorTest.java

```diff
@@ -236,4 +245,22 @@ public InetSocketAddress getSocketAddress() {
       }
     };
   }
+
+  private Collection<PartitionReplica> waitForChange(BlockingQueue<Collection<PartitionReplica>> assignmentQueue,
```

Comment:
Or if timeout passed. We could always extend the timing issue but I thought this would be add more flexibility on retry.